### PR TITLE
For test case 13

### PR DESCRIPTION
**Pull request by manabuishii**
cat4-tool.cwl, cat-job.json
Create Javascript Runtime for inputs

### CODE REVIEW

**Comment by otiai10** [comment]
Is it ok to make the `self` object as dictionary?
It it possible that `self` is accessed as an array?

**Comment by manabuishii** [comment]
Thanks

In [3.4 Parameter References](https://www.commonwl.org/v1.0/CommandLineTool.html#Parameter_references)

both case is supposed

**Comment by manabuishii** [comment]
Sorry, 

[5.2.3 CommandOutputBiding](https://www.commonwl.org/v1.0/CommandLineTool.html#CommandOutputBinding)

`self` is array.

**Comment by manabuishii** [comment]
Some conformance tests have such as

`self.basename`, `self.basename` and `self.file1`

investigate more detail.

File: input.go [hunk 1]
```diff
@@ -289,3 +291,31 @@ func (ins Inputs) Less(i, j int) bool {
 func (ins Inputs) Swap(i, j int) {
 	ins[i], ins[j] = ins[j], ins[i]
 }
+
+// CreateVM ...
+func (ins Inputs) CreateVM(srcdir string) (*otto.Otto, error) {
```
Comment (otiai10) [inline input.go]: Because this method belongs to `Inputs` struct, I prefer a name `ToJavaScriptVM` instead of `Create...`

File: input.go [hunk 2]
```diff
@@ -289,3 +291,31 @@ func (ins Inputs) Less(i, j int) bool {
...
+	self := map[string]map[string]interface{}{}
+	for _, i := range ins {
+		if i.Default != nil {
+			self[i.ID] = map[string]interface{}{
+				"path": i.Default.Entry.Location,
```
Comment (otiai10) [inline input.go]: Should we check the existence of `Entry`?

Comment (otiai10) [reply input.go]: can be null pointer

File: input.go [hunk 3]
```diff
@@ -289,3 +291,31 @@ func (ins Inputs) Less(i, j int) bool {
...
+			}
+		}
+		if i.Provided != nil {
```
Comment (otiai10) [inline input.go]: should be `early return`, or `early continue` in this case inside looping. If we prefer `Provided` than `Default`,

```
if Provided {
 // do it
 continue
}
if Default {
  // do another
  continue
}
// something if needed
```

File: input.go [hunk 4]
```diff
@@ -289,3 +291,33 @@ func (ins Inputs) Less(i, j int) bool {
 func (ins Inputs) Swap(i, j int) {
 	ins[i], ins[j] = ins[j], ins[i]
 }
+
+// CreateVM ...
+func (ins Inputs) ToJavaScriptVM(srcdir string) (*otto.Otto, error) {
```
Comment (otiai10) [inline input.go]: `srcdir` seems not used

**Review by otiai10 — CHANGES_REQUESTED** [review]

**Review by otiai10 — APPROVED** [review]

**Pull request merged**